### PR TITLE
2 0 media list

### PR DESCRIPTION
**Pull request by ngilain**
[OO-FEATURE] Add media list
https://github.com/open-orchestra/open-orchestra-media-admin-bundle/pull/339
https://github.com/open-orchestra/open-orchestra-media-bundle/pull/244
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2085
https://github.com/open-orchestra/open-orchestra/pull/1139

### CODE REVIEW

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 1]
```diff
@@ -27,62 +28,96 @@
...
-     * @Api\Groups({MediaAdminGroupContext::MEDIA_ALTERNATIVES, MediaAdminGroupContext::MEDIA_ADVANCED_LINKS})
+     * @return FacadeInterface
      */
-    public function showAction($mediaId)
+    public function listAction(Request $request)
     {
-        $media = $this->get('open_orchestra_media.repository.media')->find($mediaId);
-
-        return $this->get('open_orchestra_api.transformer_manager')->get('media')->transform($media);
+        $mapping = array(
```
Comment (alavieille) [inline MediaAdminBundle/Controller/Api/MediaController.php]: create this variable is useless

File: MediaAdminBundle/Controller/Api/MediaController.php [hunk 2]
```diff
@@ -27,62 +28,96 @@
...
+     *
+     * @Config\Route("/{mediaId}", name="open_orchestra_api_media_show")
+     * @Config\Method({"GET"})
+     *
+     * @Api\Groups({MediaAdminGroupContext::MEDIA_ALTERNATIVES, MediaAdminGroupContext::MEDIA_ADVANCED_LINKS})
+     */
+    public function showAction($mediaId)
+    {
+        $media = $this->get('open_orchestra_media.repository.media')->find($mediaId);
+
```
Comment (alavieille) [inline MediaAdminBundle/Controller/Api/MediaController.php]: check if media isn't null ?

File: MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediasView.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            let template = this._renderTemplate('List/emptyListView' , {
+                title: Translator.trans('open_orchestra_media_admin.media.title_list'),
+                urlAdd: ''
+            });
+            this.$el.html(template);
+
+            return this;
+        } else {
+            let foldersTree = new FoldersTree();
+            $.when(
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Media/MediasView.js]: usage of $.when is useless here

File: MediaAdminBundle/Resources/public/template/Navigation/Contribution/media._tpl.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <li>
-    <a href="#">
+    <a href="#media/list">
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/template/Navigation/Contribution/media._tpl.html]: use Backbone router to generate url

File: MediaAdminBundle/Resources/public/template/Media/mediasView._tpl.html [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<div class="top-action-buttons">
+    <a class="btn btn-info" href="#media/new" role="button"><%= Translator.trans('open_orchestra_backoffice.add')%></a>
```
Comment (alavieille) [inline MediaAdminBundle/Resources/public/template/Media/mediasView._tpl.html]: use Backbone router to generate url

File: MediaAdminBundle/Transformer/MediaTransformer.php [hunk 6]
```diff
@@ -33,12 +35,14 @@ public function __construct(
         FileAlternativesManager $fileAlternativesManager,
         MultiLanguagesChoiceManagerInterface $multiLanguageChoiceManager,
         $mediaDomain,
-        AuthorizationCheckerInterface $authorizationChecker
+        AuthorizationCheckerInterface $authorizationChecker,
+        MediaRepositoryInterface $mediaRepository
```
Comment (alavieille) [inline MediaAdminBundle/Transformer/MediaTransformer.php:40]: php doc

File: MediaAdminBundle/Transformer/MediaTransformer.php [hunk 7]
```diff
@@ -80,35 +77,31 @@ public function transform($mixed)
             $facade->isEditable = true;
         }
 
-        $facade->addLink('_self_select', $mixed->getId());
-        $facade->addLink('_api_full', $this->generateRoute('open_orchestra_api_media_show', array(
-            'mediaId' => $mixed->getId()
-        )));
+        $facade->addRight(
```
Comment (alavieille) [inline MediaAdminBundle/Transformer/MediaTransformer.php:81]: use a context to add rigth no ?

Comment (ngilain) [reply MediaAdminBundle/Transformer/MediaTransformer.php:81]: @alavieille : It's not required as the right are always required for now

**Comment by ngilain** [comment]
@alavieille : update done

File: MediaAdminBundle/Transformer/MediaTransformer.php (hunk 7 above)
Comment (alavieille) [reply MediaAdminBundle/Transformer/MediaTransformer.php:81]: ok

**Pull request merged**